pkg/macaroon/apikey.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package macaroon
+
+import (
+	"bytes"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/gogo/protobuf/proto"
+	"github.com/zeebo/errs"
+)
+
+var (
+	// Error is a general API Key error
+	Error = errs.Class("api key error")
+	// ErrFormat means that the structural formatting of the API Key is invalid
+	ErrFormat = errs.Class("api key format error")
+	// ErrInvalid means that the API Key is improperly signed
+	ErrInvalid = errs.Class("api key invalid error")
+	// ErrUnauthorized means that the API key does not grant the requested permission
+	ErrUnauthorized = errs.Class("api key unauthorized error")
+	// ErrRevoked means the API key has been revoked
+	ErrRevoked = errs.Class("api key revocation error")
+)
+
+// APIKey implements a Macaroon-backed Storj-v3 API key.
+type APIKey struct {
+	mac *Macaroon
+}
+
+// ParseAPIKey parses a given api key string and returns an APIKey if the
+// APIKey was correctly formatted. It does not validate the key.
+func ParseAPIKey(key string) (*APIKey, error) {
+	data, version, err := base58.CheckDecode(key)
+	if err != nil || version != 0 {
```

Comment:
why do you want the version to be `0` ?

Reply:
if we ever change the version, this code doesn't know what to do, so i want it to fail